src/main/java/com/craftingguide/exporter/extensions/minecraft/OtherRecipeGatherer.java

```diff
@@ -0,0 +1,88 @@
+
+import net.minecraft.item.ItemStack;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.SortedSet;
+
+import codechicken.nei.PositionedStack;
+import codechicken.nei.recipe.GuiUsageRecipe;
+import codechicken.nei.recipe.IUsageHandler;
+
+public class OtherRecipeGatherer extends Gatherer {
+	
+	@Override
+	public void gather(ModPackModel modpack) {
+		ArrayList<IUsageHandler> machines = GuiUsageRecipe.usagehandlers;
+		Iterator<ItemModel> allItems = modpack.getAllItems().iterator();
+		ArrayList<ItemModel> alreadyRegisteredMachines = new ArrayList<ItemModel>();
+		while (allItems.hasNext()) {
+			ItemModel item = allItems.next();
+			SortedSet<RecipeModel> recipes = item.getRecipes();
+			for (RecipeModel recipe : recipes) {
+				if (!recipe.getTools().isEmpty()) {
+					if (!alreadyRegisteredMachines.contains(recipe.getTools().first()))
+					alreadyRegisteredMachines.add(recipe.getTools().first());
+				}
+			}
+		}
+		HashMap<String, ItemStack> namedItems = new HashMap<String, ItemStack>();
+		for (IUsageHandler machine : machines) {
+			boolean alreadyRegistered = false;
+			for (ItemModel registeredMachine : alreadyRegisteredMachines) {
+				if (machine.getRecipeName() == registeredMachine.getRawItemStack().getDisplayName()) {
+					alreadyRegistered = true;
+					break;
+				}
+			}
+			if (alreadyRegistered) continue;
+			boolean notRealItem = true;
+			ItemModel item = null;
+			Iterator<ItemModel> allItemIterator = modpack.getAllItems().iterator();
+			if (!namedItems.containsKey(machine.getRecipeName())) {
+				while (allItemIterator.hasNext()) {
+					ItemModel oneItem = allItemIterator.next();
+					if (oneItem.getDisplayName() == machine.getRecipeName()) {
+						notRealItem = false;
+						item = oneItem;
+						break;
+					}
+				}
+				if (notRealItem) {
+					continue;
+				}
+			}
+			for (int recipe = 0; recipe < machine.numRecipes(); recipe++) {
+				ItemStack result = machine.getResultStack(recipe).item;
+				List<PositionedStack> ingrediants = machine.getIngredientStacks(recipe);
+				ArrayList<ItemStack> ingrediantStacks = new ArrayList<ItemStack>();
+				for (PositionedStack ingrediant : ingrediants) {
+					ingrediantStacks.add(ingrediant.item);
+				}
+				ArrayList<ItemStack> otherStacks = new ArrayList<ItemStack>();
+				List<PositionedStack> others = machine.getOtherStacks(recipe);
+				for (PositionedStack other : others) {
+					otherStacks.add(other.item);
+				}
+				RecipeModel recipeModel = new RecipeModel(new ItemStackModel(modpack.getItem(result), result.stackSize));
+				for (ItemStack input : ingrediantStacks) {
+					recipeModel.addInput(new ItemStackModel(modpack.getItem(input), input.stackSize));
+				}
+				for (ItemStack extra : otherStacks) {
+					recipeModel.addInput(new ItemStackModel(modpack.getItem(extra), extra.stackSize));
+				}
+				item.addRecipe(recipeModel);
+			}
+		}
+	}
+	
+}
```

Comment:
This method is much too large.  You'll notice all the methods in the rest of the classes are factored into small chunks with very clear names: many of them private.  Please try to do the same here.  Trust me... large methods like this rapidly become very, very large, and very, very impossible to maintain.

Also, a few stylistic notes:

  * watch your spelling: ingredient, not ingrediant
  * indentation should be 4 *spaces*
  * for loop indices, use names like "index" or "currentRecipeIndex" or such.  Save names like "recipe" for referring to objects of that type.  Makes it much easier to remember what each thing is.
  * when declaring variables, prefer the more generic type (e.g., `Map` instead of `HashMap`).  Save the specific type only for when you create instances.  That way, if you need to change your mind later, you only have to change the one place.
  * it's generally preferable to declare iterators within the loop construct rather than creating a variable to contain them (e.g., `for (ItemModel item : modPack.getAllItems())`.  Creates less code, and fewer variables to think about.